kartothek/io/eager.py

```diff
@@ -497,6 +502,9 @@ def store_dataframes_as_dataset(
         store=store, dataset_uuid=dataset_uuid, df_serializer=df_serializer
     )
 
+    if secondary_indices:
```

Comment:
you should build indices before you `store_dataframes`, as you require the data to be present at the metapartition (it is gone once you have stored it)

Reply:
oh ya! Thanks